src/bin.ts

```diff
@@ -566,12 +623,13 @@ function phase4(payload: BootstrapState) {
 
   // Prepend `ts-node` arguments to CLI for child processes.
   process.execArgv.push(
-    entrypoint,
+    tsNodeScript,
     ...argv.slice(2, argv.length - restArgs.length)
   );
+
   // TODO this comes from BoostrapState
   process.argv = [process.argv[1]]
-    .concat(executeEntrypoint ? ([scriptPath] as string[]) : [])
+    .concat(executeEntrypoint ? ([entryPointPath] as string[]) : [])
     .concat(restArgs.slice(executeEntrypoint ? 1 : 0));
```

Comment:
`executeEntrypoint` is computed from our `parseArgvResult` and `phase2Result`.  Is it possible that `executeEntrypoint` might be false in a child process, when actually, it should be true?  Because the host was a REPL, but the child should not be?

I'm not sure, but I'm wondering if there is a situation where someone uses our REPL to `require('./this-script-will-fork-a-child-process')`

In that case, the host process has `executeEntrypoint === false` because it's a REPL.  But the forked child should have it set to true.

I see in `getEntryPointInfo`, we are using `argvResult.code`, `.interactive`, and `.restArgs`  In a forked process, should we avoid doing that?

Reply:
Nevermind, I see now that `restArgs` is overwritten in `child-entrypoint.ts`.

Reply:
Nevermind, I see now that `restArgs` is overwritten in `child-entrypoint.ts`.